sky/server/server.py

```diff
@@ -164,8 +166,53 @@ async def dispatch(self, request: fastapi.Request, call_next):
         return await call_next(request)
 
 
+def is_browser_request(request: fastapi.Request) -> bool:
+    """Checks if a request is made by a browser like user agent."""
+    return request.headers['User-Agent'].startswith('Mozilla')
+
+
+class CacheControlStaticMiddleware(starlette.middleware.base.BaseHTTPMiddleware
+                                  ):
+    """Middleware to add cache control headers to static files."""
+
+    async def dispatch(self, request: fastapi.Request, call_next):
+        if request.url.path.startswith('/dashboard/_next'):
+            response = await call_next(request)
+            response.headers['Cache-Control'] = 'max-age=86400'
+            return response
+        return await call_next(request)
+
+
+class BrowserNoPostPutMiddleware(starlette.middleware.base.BaseHTTPMiddleware):
+    """Middleware to prevent POST and PUT requests from browsers."""
+
+    async def dispatch(self, request: fastapi.Request, call_next):
+        if request.url.path.startswith('/dashboard/') and is_browser_request(
+                request) and request.method in ['POST', 'PUT']:
+            raise fastapi.HTTPException(status_code=405,
+                                        detail='Method not allowed')
+        return await call_next(request)
```

Comment:
Why is this needed?

Reply:
This is meant to reject post and put requests from browser directly, after double-check the handler, we have defined `get` method, I will remove this.